oxide-auth/src/primitives/registrar.rs

```diff
@@ -291,107 +286,69 @@ pub trait PasswordPolicy: Send + Sync {
     fn check(&self, client_id: &str, passphrase: &[u8], stored: &[u8]) -> Result<(), RegistrarError>;
 }
 
-/// Store passwords using `Pbkdf2` to derive the stored value.
-///
-/// Each instantiation generates a 16 byte random salt and prepends this additionally with the
-/// username. This combined string is then used as the salt using the passphrase as the secret to
-/// derive the output. The iteration count defaults to `65536` but can be customized.
-pub struct Pbkdf2 {
-    /// A prebuilt random, or constructing one as needed.
-    random: Option<SystemRandom>,
-    iterations: NonZeroU32,
-}
-
-impl Default for Pbkdf2 {
-    fn default() -> Self {
-        Pbkdf2 {
-            random: Some(SystemRandom::new()),
-            .. *PBKDF2_DEFAULTS
-        }
-    }
+/// Store passwords using `Argon2` to derive the stored value.
+#[derive(Debug)]
+pub struct Argon2<'a> {
+    secret_key: SecretKey<'a>
 }
 
-impl Clone for Pbkdf2 {
-    fn clone(&self) -> Self {
-        Pbkdf2 {
-            random: Some(SystemRandom::new()),
-            .. *self
+impl<'a> Default for Argon2<'a> {
+    fn default() -> Self {   
+        Argon2 {
+            secret_key: "Please provide a SecretKey! This is for your own protection.".into()
         }
     }
 }
 
-impl fmt::Debug for Pbkdf2 {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        f.debug_struct("Pbkdf2")
-            .field("iterations", &self.iterations)
-            .field("random", &())
-            .finish()
+impl From<std::string::FromUtf8Error> for RegistrarError {
```

Comment:
Those impls are part of the crate's interface. I find this one a bit confusing (although it might be useful in general).

Reply:
I will remove both impl